keras/utils/io_utils.py

```diff
@@ -207,6 +209,23 @@ def is_path_instance(path):
                             'Received: {}.'.format(type(path)))
         self.read_only = mode == 'r'
 
+    @staticmethod
+    def is_supported_type(path):
+        """Check if `path` is of supported type for instantiating a `H5Dict`"""
+        return (
+            isinstance(path, h5py.Group) or
+            isinstance(path, dict) or
+            isinstance(path, six.string_types) or
+            _is_path_instance(path)
+        )
+
+    @staticmethod
+    def opens_file(path):
```

Comment:
I think this has the same role as `is_supported_type` just above? The idea was to put those methods as members of the class as they say something about what happens when instantiating specifically the`H5Dict` with a given argument, used as:
```
H5Dict.is_supported_type(arg)
H5Dict.opens_file(arg)
```
Which I think is pretty clear. If it's moved out, maybe it should be renamed `makes_h5dict_open_file` if you see what I mean?

Reply:
Actually, the whole thing is a bit silly, the only thing we want to do is to check if `path` is something that is interpreted as a file path - in a python version independent way. How about we add:
```
def accepted_as_path(path):
    if isinstance(path, six.string_types):
        return True

    try:
        import pathlib
        if isinstance(path, pathlib.Path):
            return True
    except ImportError:
        pass

    try:
        import pathlib2
        if isinstance(path, pathlib2.Path):
            return True
    except ImportError:
        pass

    return False
```
to `io_utils` and remove the current (hacky) `_is_path_instance` function? @fchollet

Reply:
Or I guess this is the cleaner (version independent) way to do this:
```
def is_path_like(path):
    return hasattr('__fspath__') and callable(path.__fspath__)
```
However, it currently won't work to use a `os.PathLike` object anyway, because `os.path.isfile` (which is used in conjunction) wont accept it in Python 2  - let's solve this in a separate PR, it has nothing to do with the additions in this PR anyway. I removed the `opens_file` now as well and just use the simpler `opens_file = not isinstance(filepath, (dict, h5py.Group))`